imsim/stamp.py

```diff
@@ -84,6 +84,27 @@ def setup(self, config, base, xsize, ysize, ignore, logger):
         Returns:
             xsize, ysize, image_pos, world_pos
         """
+
+        # First do a parsing check to make sure that all the options passed to
+        # the config are valid, and no required options are missing.
+
+        req = {'fft_sb_thresh': float,
+               'band': str,
+               'world_pos': galsim.CelestialCoord,
+               'diffraction_psf': dict
+               }
+
+        opt = {'airmass': dict,
+               'rawSeeing': float,
+               'max_flux_simple': float,
+               'draw_method': str,
+               'maxN': int,
+               'size': int,
+               'photon_ops': dict
+               }
+
+        galsim.config.CheckAllParams(config, req, opt)
```

Comment:
Since I think we've converted everything to optional, you can more simply just call
```
ignore = ['fft_sb_thresh', 'band', 'airmass', 'rawSeeing', 'max_flux_simple'] + ignore
galsim.config.CheckAllParams(config, ignore=ignore)
```
If only calling Check, rather than Get, things in ignore are effectively treated the same as things in opt.

Reply:
I think we still currently need the diffraction_psf as required.